embeddings/task/sklearn_task/text_classification.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict, Optional
+
+from numpy import typing as nptyping
+from sklearn.base import ClassifierMixin as AnySklearnClassifier
+
+from embeddings.embedding.sklearn_embedding import SklearnEmbedding
+from embeddings.task.sklearn_task.sklearn_task import SklearnTask
+from embeddings.utils.array_like import ArrayLike
+
+
+class TextClassification(SklearnTask):
+    def __init__(
+        self,
+        classifier: AnySklearnClassifier,
+        classifier_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__()
+        self.classifier_kwargs = classifier_kwargs if classifier_kwargs else {}
+        self.classifier = classifier(**self.classifier_kwargs)
+
+    def build_task_model(self, embedding: SklearnEmbedding) -> None:
+        self.embedding = embedding
+
+    def fit(
+        self,
+        x_train: ArrayLike,
+        y_train: ArrayLike,
+    ) -> None:
+        self.classifier.fit(x_train, y_train)
+
+    def predict(
+        self,
+        data: Dict[str, ArrayLike],
+        predict_subset: str = "test",
+    ) -> Dict[str, nptyping.NDArray[Any]]:
+        predictions = self.classifier.predict(self.embedding.embed(data[predict_subset]["x"]))
+        model_result = {
+            "y_pred": predictions,
+            "y_true": data[predict_subset]["y"].values,
+        }
+        return model_result
+
+    def fit_predict(
+        self,
+        data: Dict[str, ArrayLike],
+        predict_subset: str = "test",
+    ) -> Dict[str, nptyping.NDArray[Any]]:
+        x_train = self.embedding.embed(data["train"]["x"])
+        y_train = data["train"]["y"]
+        self.fit(x_train, y_train)
+        return self.predict(data)
```

Comment:
one more suggestion here, I'd unify the way we call fit and predict methods. If we extract x_train and y_train for the fit method I'd do the same for predict method or the other way round

```suggestion
        x_train = self.embedding.embed(data["train"]["x"])
        y_train = data["train"]["y"]
        self.fit(x_train, y_train)
        x_predict = self.embedding.embed(data[predict_subset]["x"])
        y_predict = self.predict(x_predict)
        return {
            "y_pred": y_predict,
            "y_true": data[predict_subset]["y"].values,
        }
```
or
```suggestion
        self.fit(data)
        return self.predict(data)
```

but it's up to you and you can leave it as it is

Reply:
I think the second option looks better.